node/cli/src/command.rs

```diff
@@ -418,26 +404,34 @@ pub fn run() -> Result<()> {
 				_ => panic!("invalid chain spec"),
 			}
 		}
-		Some(Subcommand::ExportGenesisHead(params)) => {
+		Some(Subcommand::ExportGenesisHead(cmd)) => {
+			let runner = cli.create_runner(cmd)?;
+			let chain_spec = runner.config().chain_spec.cloned_box();
+
 			let mut builder = sc_cli::LoggerBuilder::new("");
 			builder.with_profiling(sc_tracing::TracingReceiver::Log, "");
 			let _ = builder.init();
 
-			// Cumulus approach here, we directly call the generic load_spec func
-			let chain_spec = load_spec(
-				params.chain.as_deref().unwrap_or_default(),
-				params.parachain_id.unwrap_or(1000).into(),
-				&cli.run,
-			)?;
-			let state_version = Cli::runtime_version(&chain_spec).state_version();
+			let storage = chain_spec.build_storage()?;
+			let executor = sc_executor::WasmExecutor::<HostFunctions>::builder()
+				.with_execution_method(sc_executor::WasmExecutionMethod::Compiled {
+					instantiation_strategy: WasmtimeInstantiationStrategy::PoolingCopyOnWrite,
+				})
+				.with_max_runtime_instances(runner.config().executor.max_runtime_instances)
+				.build();
+
+			let state_version = sc_chain_spec::resolve_state_version_from_wasm::<
+				_,
+				HashingFor<Block>,
+			>(&storage, &executor)?;
```

Comment:
When running `moonbeam export-genesis-state --chain moonriver`, the expected `state_version` is `StateVersion::V1`, but it returns `StateVersion::V0`. 

I think the issue is on the `executor`.